### PR TITLE
Bitcoin volume chart

### PR DESCRIPTION
**Pull request by cbrainy01**

### CODE REVIEW

File: src/components/BitcoinOverview/BitcoinOverview.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import VolumeChart from 'components/VolumeChart';
+import { StyledBitcoinOverview } from './BitcoinOverview.styles';
+
+export class BitcoinOverview extends React.Component {
+  render() {
+    return (
+    <StyledBitcoinOverview>
+        <h1>Bitcoin Overview</h1>
+        <PriceChart currency={ this.props.currency }/>
+        <VolumeChart currency={ this.props.currency } />
```
Comment (yoloOnTheBattlefield) [inline src/components/BitcoinOverview/BitcoinOverview.js]: currency={this.props.currency}

File: src/components/BitcoinOverview/BitcoinOverview.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import PriceChart from 'components/PriceChart';
+import VolumeChart from 'components/VolumeChart';
```
Comment (yoloOnTheBattlefield) [inline src/components/BitcoinOverview/BitcoinOverview.js]: import { PriceChart, VolumeChart}  from 'components';

File: src/components/Navbar/Navbar.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import axios from "axios";
-import NavbarGlobal from "components/NavbarGlobal/NavbarGlobal";
+import NavbarGlobal from "components/NavbarGlobal";
```
Comment (yoloOnTheBattlefield) [inline src/components/Navbar/Navbar.js]: import { NavbarGlobal } from "components";

File: src/components/PriceChart/PriceChart.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import axios from "axios"
+import Chart from "chart.js/auto";
+import { Line } from "react-chartjs-2"
+import wordedDate from 'utils/wordedDate';
+import { StyledPriceChart } from './PriceChart.styles'
```
Comment (yoloOnTheBattlefield) [inline src/components/PriceChart/PriceChart.js]: relative should come after absolute imports

File: src/components/VolumeChart/VolumeChart.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import axios from "axios";
+import Chart from "chart.js/auto";
+import { Bar } from "react-chartjs-2";
+import wordedDate from "utils/wordedDate";
+import { StyledVolumeChart } from './VolumeChart.styles'
```
Comment (yoloOnTheBattlefield) [inline src/components/VolumeChart/VolumeChart.js:6]: relative should come after absolute imports

File: src/pages/CoinList/CoinList.js [hunk 6]
```diff
@@ -1,10 +1,13 @@
 import React from "react";
-
+import BitcoinOverview from "components/BitcoinOverview/index";
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinList/CoinList.js]: import { BitcoinOverview  }from "components";

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/components/VolumeChart/VolumeChart.js (hunk 5 above)
Comment (cbrainy01) [reply src/components/VolumeChart/VolumeChart.js:6]: I thought library imports didn't follow that rule.

File: src/App.js [hunk 7]
```diff
@@ -1,27 +1,38 @@
...
 import './App.css';
 
 class App extends React.Component {
-
   state = {
     currency: "usd"
   }
 
+  componentDidMount() {
+    const currency = localStorage.getItem("currency", "usd") || "usd"
```
Comment (yoloOnTheBattlefield) [inline src/App.js]: getItem only requires one argument

File: src/components/BitcoinOverview/BitcoinOverview.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+export class BitcoinOverview extends React.Component {
+    state={
+        timeSpan: 1
+    }
+  
+    handleChange = (e) => { this.setState({timeSpan: e.target.value}) }
+
+    render() {
+    return (
```
Comment (yoloOnTheBattlefield) [inline src/components/BitcoinOverview/BitcoinOverview.js]: check your indentations

File: src/components/Navbar/Navbar.js [hunk 9]
```diff
@@ -39,15 +39,15 @@ export default class Navbar extends React.Component {
     render() {
         return(
             <StyledNavbar>
-                <StyledLink to="/">CoinList</StyledLink>
+                <StyledLink to={`/`}>CoinList</StyledLink>
```
Comment (yoloOnTheBattlefield) [inline src/components/Navbar/Navbar.js]: `to="/"`

File: src/components/NavbarGlobal/NavbarGlobal.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import { StyledNavbarGlobal } from './NavbarGlobal.styles'
-import { formatNumber } from 'utils/formatNumber'
+import { formatNumber } from 'utils'
```
Comment (yoloOnTheBattlefield) [inline src/components/NavbarGlobal/NavbarGlobal.js:2]: absolute above relative

File: src/components/PriceChart/PriceChart.js [hunk 11]
```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import axios from "axios";
+import Chart from "chart.js/auto";
+import wordedDate from 'utils/wordedDate';
+import { Line } from "react-chartjs-2";
+import { StyledPriceChart } from './PriceChart.styles';
+import { getPreviousDates, formatOverviewNumber, startDate } from "utils"
```
Comment (yoloOnTheBattlefield) [inline src/components/PriceChart/PriceChart.js]: absolute before relative

File: src/components/PriceChart/PriceChart.js [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+          
+          function span(timeSpan) {
+            if(timeSpan === "1week") { return "1week" }
+            else if(timeSpan == 1) { return 30 }
+            else if(timeSpan == 3) { return 90}
+            else if(timeSpan == 6) { return 180}
+            else if(timeSpan == 12) { return 360}
+          }
+            
+            const response = await axios(
```
Comment (yoloOnTheBattlefield) [inline src/components/PriceChart/PriceChart.js:38]: check your indentation

File: src/components/PriceChart/PriceChart.js [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+        this.getMarketChartData()
+    }
+    componentDidUpdate(prevProps) {
+        if(prevProps.currency !== this.props.currency || prevProps.timeSpan !== this.props.timeSpan) {
+            this.getMarketChartData()
+        }
+    }
+
+    render() {
+
```
Comment (yoloOnTheBattlefield) [inline src/components/PriceChart/PriceChart.js]: new line not needed

File: src/components/PriceChart/PriceChart.styles.js [hunk 14]
```diff
@@ -0,0 +1,3 @@
+import styled from "styled-components";
+
```
Comment (yoloOnTheBattlefield) [inline src/components/PriceChart/PriceChart.styles.js]: new line not needed

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 15]
```diff
@@ -1,30 +1,42 @@
-import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import React from "react";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import { Navbar } from "components";
 import {CoinList, CoinPage, Portfolio} from "pages";
```
Comment (yoloOnTheBattlefield) [inline src/App.js]: add spaces between curly braces

File: src/App.js [hunk 16]
```diff
@@ -1,30 +1,42 @@
...
 
+  componentDidMount() {
+    const currency = localStorage.getItem("currency") || "usd"
+    this.setState({currency: currency})
+  }
+  
   handleCurrencyChange = (newCurrency) => {
-    this.setState({ currency: newCurrency })
+    localStorage.setItem("currency", newCurrency)
+    this.setState({currency: newCurrency})
```
Comment (yoloOnTheBattlefield) [inline src/App.js]: add spaces between curly braces

File: src/components/NavbarGlobal/NavbarGlobal.js (hunk 10 above)
Comment (yoloOnTheBattlefield) [reply src/components/NavbarGlobal/NavbarGlobal.js:2]: yo

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**